client/src/components/ChatMessage.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Message } from '../types';
+
+interface Props {
+  id: string;
+  text: string;
+}
+
+const ChatMessage: React.FunctionComponent<Props> = ({ id, text }) => {
+  return <div>{this.prop.text}</div>;
```

Comment:
`this` doesn't work in function arrow components ([some interesting reading](https://www.codementor.io/dariogarciamoya/understanding-this-in-javascript-with-arrow-functions-gcpjwfyuc)). Instead, we can just reference text directly:

`return <div>{text} </div>`

This is because the component is just a function that takes in a props object and returns a React component.

```
const ChatMessage: React.FunctionComponent<Props> = (props) => {
   // ES6 destructuring, equivalent to
   // const id = props.id
   // const text = props.text
  const { id, text } = props;
  return <div>{text}</div>;
```

Reply:
Oh yes, I remember reading about destructuring. Didn't recognize here. Must've forgot. Thanks for pointing that out!